### DISCUSSION TITLE
I have a question about the version of fsspec

### DISCUSSION THREAD

**Discussion by karajan1001** [fsspec/filesystem_spec#1239] [General]
Excuse me, I'm the author of ossfs, coming from a ticket from a user.
https://github.com/fsspec/ossfs/issues/76

Actually, in previous, I never followed the version of fsspec's step just released it as I wish. But it creates some confusion to the users. So I want to know if there are any general rules for the versions of fsspec. In my opinion, the best practice is to make all fss synced to the same version. But if we do this, some low-frequency updated fs like ossfs might need to release a new version even if no real changes are made to it (except updating the dependency of fsspec)? On another side, sometimes fsspec might not have a release plan in the near future, but this time I had made an important change to ossfs, what should I do in this case? My suggestion is to increase the small number (2023.3.0 -> 2023.3.1). But it's not intuition for the users because, in most cases, the small number corresponds to a small change. 

Thank you

**Comment by martindurant** [comment]
fsspec does not have a strict release cadence, but we do commonly have about one release per month. The situation describe is true for the closely related repos s3fs and gcsfs, which are always released at the same time and with the same version ID and exact version pin to fsspec, whether they have had changes since the last release or not. Other downstream implementations probably do not need to be this strict, especially if they are not async (which is always more complicated).

> in most cases, the small number corresponds to a small change

Our versioning now is strictly time-based, and the major/minor/patch notion is not used. The last digit is just incremented; so the first release in April would be 2023.4.0, the second 2023.4.1, etc. 

If you need a release for a compatible change, please just ping here. We can agree on a version tag, even if the exact release date ends up drifting slightly into the next month.

**Comment by martindurant** [comment]
Of course, ossfs might choose not to have an exact pin on fsspec and sidestep this issue. It means that old releases could then be incompatible with and broken by newer fsspec; but this might be acceptable to your users.

**Comment by karajan1001** [comment]
Thank you @martindurant, I would try to fix the version of `ossfs` with `fsspec`, and if there is a compatible change I will pin you.
